### DISCUSSION TITLE
onBlur is called with old table state

### DISCUSSION THREAD

**Discussion by DonHaul** [nick-keller/react-datasheet-grid#190] [Q&A] [answered]
I am encountering this strange behaviour where if I try to access the table data, inside the on the `onBlur` callback, I get an old version of the table data.

see here:
https://codesandbox.io/s/cranky-leftpad-ofgbzt?file=/src/App.js

Both the Button's `onClick` callback and the Table's `onBlur` callback call the same function, while one returns the table current state (1 row of data) the other shows 0 rows of data (initial state) any idea of what might be happening?  

See image of result of clicking button vs triggering the `onBlur` callback
![image](https://user-images.githubusercontent.com/13081142/192108389-2991a3c3-a4df-4311-8fc6-d058caf31fc2.png)

EDIT: This behavior seems to only occur in DataSheetGrid not in DynamicDataSheetGrid . Is this intended?

**Comment by nick-keller** [comment, marked as answer]
Hello @DonHaul,
Yes this is intended. As you [may know](https://react-datasheet-grid.netlify.app/docs/performance/static-vs-dynamic), the default version of DSG is static, and the static version "saves the props it receives on the first render and never updates them".

This means that the `onBlur` function that is called is the one you pass on the very first render. And it displays the `data` that is had when the function was created, here the `{}` from `const [data, setData] = useState({})`
